### PR TITLE
feat: add redis connection, update scheduler deps

### PR DESCRIPTION
**Pull request by ServiceDog**
Adds a Redis connection to:
- Store itineraries and their calculated price details in memory with auto-expiration
- Provide a draft ID to the customer to confirm without being able to alter the itinerary details

depends on https://github.com/aetheric-oss/svc-scheduler/pull/48

### CODE REVIEW

**Comment by ServiceDog** [comment]
moving this back to draft, some churn with the dependencies

File: log4rs.yaml [hunk 1]
```diff
@@ -62,6 +76,10 @@ loggers:
     level: info
     appenders:
       - rest_requests
+  app::cache:
```
Comment (owlot) [inline log4rs.yaml:79]: This should be `backend::cache` as it depends on the redis backend to be running. It should append to the `backend_requests` appender

File: log4rs.yaml [hunk 2]
```diff
@@ -33,6 +33,20 @@ appenders:
         base: 1
     encoder:
       kind: json
+  cache:
```
Comment (owlot) [inline log4rs.yaml:36]: In other services that use redis (like svc-telemetry) we have a `backend_requests` appender.

File: server/src/cache/macros.rs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+//! log macro's for cache logging
+
+use lib_common::log_macros;
+log_macros!("cache", "app::cache");
```
Comment (owlot) [inline server/src/cache/macros.rs:4]: should be `backend::cache` instead of `app::cache`

Comment (ServiceDog) [reply server/src/cache/macros.rs]: https://github.com/aetheric-oss/svc-cargo/pull/43/commits/5eccc2bc4b4335866a129e31c449d470990edf83

File: log4rs.yaml (hunk 2 above)
Comment (ServiceDog) [reply log4rs.yaml]: https://github.com/aetheric-oss/svc-cargo/pull/43/commits/5eccc2bc4b4335866a129e31c449d470990edf83

File: log4rs.yaml (hunk 1 above)
Comment (ServiceDog) [reply log4rs.yaml]: https://github.com/aetheric-oss/svc-cargo/pull/43/commits/5eccc2bc4b4335866a129e31c449d470990edf83

**Pull request merged**